recipes/microservice-essentials/all/conanfile.py

```diff
@@ -0,0 +1,104 @@
+    topics = ("microservices", "cloud-native", "request-handling")
+    settings = "os", "compiler", "arch", "build_type"    
+    package_type = "library"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_tests": [True, False],
+        "with_examples": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_tests": False,
+        "with_examples": False
+    }
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+            "Visual Studio": "15.7",
+        }
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.16.3 <4]")
+
+    def requirements(self):
+        if self.options.with_examples:
+            self.requires("cpp-httplib/0.12.4")
+            self.requires("nlohmann_json/3.11.2")
+            self.requires("openssl/[>=3 <4]")
+            self.requires("grpc/1.50.1")
+        if self.options.with_tests:
+            self.requires("catch2/3.3.2")
+            self.requires("nlohmann_json/3.11.2")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
+        tc.variables["BUILD_TESTING"] = self.options.with_tests
+        tc.variables["BUILD_EXAMPLES"] = self.options.with_examples
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, "17")
+
+        def loose_lt_semver(v1, v2):
+            lv1 = [int(v) for v in v1.split(".")]
+            lv2 = [int(v) for v in v2.split(".")]
+            min_length = min(len(lv1), len(lv2))
+            return lv1[:min_length] < lv2[:min_length]
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and loose_lt_semver(str(self.settings.compiler.version), minimum_version):
+            raise ConanInvalidConfiguration(
+                "{} requires C++17, which your compiler does not support.".format(self.name)
+            )
+    
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def package(self):
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        copy(self, "*.dll", src=self.source_folder, dst=os.path.join(self.package_folder, "bin"), keep_path=False)
```

Comment:
I think you need this because probably the CMakeLists is installing the `dll's` to other place that's not `bin` (maybe also to `lib`?) which is the default, so in the end the dll's will end up in two places, the ones copied by `cmake.install()` and the ones by the `copy()`. You an set the `cpp_info.bindirs = ["lib"]` if that's the case but the expected thing would be that the dll's end up in the `bin` folder

Reply:
CMake installs the dlls to the lib folder. That can be changed some time in the future but not for that initial 0.1.0 version. Other recipes use CMakeLists.txt patches for the same issue. If possible, I'd like to avoid that. Thus, I updated the .bindirs property. Maybe not perfect, but better than deploying two copies of the dlls.